shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/providers/TemplateProviderImpl.java

```diff
@@ -0,0 +1,28 @@
+package com.redhat.service.bridge.shard.operator.providers;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.client.utils.Serialization;
+
+@ApplicationScoped
+public class TemplateProviderImpl implements TemplateProvider {
+    private static final String TEMPLATES_DIR = "/templates/";
+    private static final String INGRESS_DEPLOYMENT_PATH = TEMPLATES_DIR + "bridge-ingress-deployment.yaml";
+
+    @Override
+    public Deployment loadIngressDeploymentTemplate() {
+        return loadYaml(Deployment.class, INGRESS_DEPLOYMENT_PATH);
```

Comment:
Interesting! Maybe we can consider using Qute in the future instead if we want to have a more "verbose" template file. Just an idea for the future in case we need it.